tests/foreman/cli/test_satellitesync.py

```diff
@@ -2022,6 +2023,174 @@ def test_positive_export_import_incremental_yum_repo(
+            function_sca_manifest_org, PULP_EXPORT_DIR
+        )
+
+        # On the importing side import version 1, check the package count.
+        import_path1 = target_sat.move_pulp_archive(function_sca_manifest_org, export_1['message'])
+        target_sat.cli.ContentImport.version(
+            {'organization-id': function_import_org_with_manifest.id, 'path': import_path1}
+        )
+        imp_cv = target_sat.cli.ContentView.info(
+            {'name': exp_cv['name'], 'organization-id': function_import_org_with_manifest.id}
+        )
+        assert len(imp_cv['versions']) == 1
+        imp_cvv = imp_cv['versions'][0]
+        assert target_sat.api.ContentViewVersion(id=imp_cvv['id']).read().package_count == pkg_cnt_1
+
+        # Create an AK with the imported CV, register the content host and check
+        # the package count available to install. Filtered package should be missing.
+        ak = target_sat.cli_factory.make_activation_key(
+            {
+                'content-view': exp_cv['name'],
+                'lifecycle-environment': ENVIRONMENT,
+                'organization-id': function_import_org_with_manifest.id,
+            }
+        )
+        target_sat.cli.ActivationKey.content_override(
+            {
+                'id': ak.id,
+                'content-label': function_synced_rhel_repo['content-label'],
+                'value': 'true',
+            }
+        )
+        res = rhel_contenthost.register(
+            function_import_org_with_manifest, None, ak.name, target_sat
+        )
+        assert res.status == 0, (
+            f'Failed to register host: {rhel_contenthost.hostname}\n'
+            f'StdOut: {res.stdout}\nStdErr: {res.stderr}'
+        )
+        assert rhel_contenthost.subscribed
+        res = rhel_contenthost.execute('dnf clean all && dnf repolist -v')
+        assert res.status == 0
+        assert (
+            f'Repo-available-pkgs: {pkg_cnt_1}' in res.stdout
+        ), 'Package count available on the host did not meet the expectation'
+
+        res = rhel_contenthost.execute(f'dnf -y install {filtered_pkg}')
+        assert res.status, 'Installation of filtered package succeeded unexpectedly'
+        assert f'No match for argument: {filtered_pkg}' in res.stdout
+
+        # Update the fiter so that no package is left behind, publish version 2 and export it.
+        target_sat.cli.ContentView.filter.rule.update(
+            {
+                'content-view-filter-id': cvf['filter-id'],
+                'id': cvf_rule['rule-id'],
+                'name': gen_string('alpha'),
+            }
+        )
+        target_sat.cli.ContentView.publish({'id': exp_cv['id']})
+        exp_cv = target_sat.cli.ContentView.info({'id': exp_cv['id']})
+        assert len(exp_cv['versions']) == 2
+        cvv_2 = max(exp_cv['versions'], key=lambda x: int(x['id']))
+        pkg_cnt_2 = target_sat.api.ContentViewVersion(id=cvv_2['id']).read().package_count
```

Comment:
```suggestion
        pkg_cnt_2 = target_sat.api.ContentViewVersion(id=cvv_2['id']).read().package_count
        assert pkg_cnt_2 > pkg_cnt_1
```

Reply:
We can add this.

Reply:
This might be nitpicky, but `assert pkg_cnt_2 == pkg_cnt_1 + 1` catches more.

Reply:
given the reasoning [here](https://github.com/SatelliteQE/robottelo/pull/13202#discussion_r1456231753) i am happy with the @lhellebr 's suggestion